src/include/sof/audio/codec_adapter/codec/generic.h

```diff
@@ -188,6 +188,8 @@ struct processing_module {
 	struct comp_buffer *ca_source;
 	struct comp_buffer *local_buff;
 	struct sof_ipc_stream_params stream_params;
+	struct module_interface *interface;
+	struct comp_dev *dev;
```

Comment:
Can you explain what is the purpose of added fields?

Reply:
I suspect we are going to have some transitional data until the work is complete, but comp_dev will be the low level interface to pipeline for the moment.

Reply:
I will drop "interface" as it is used only in the wrapper. But comp_dev is needed for all trace messages right now. Eventually, I will modify the trace message to add traces for modules and not need comp_dev/. But it will have to be a separate PR